trace-utils/src/stats_utils.rs

```diff
@@ -0,0 +1,82 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+use flate2::{write::GzEncoder, Compression};
+use hyper::{body::Buf, Body, Client, Method, Request, StatusCode};
+use hyper_rustls::HttpsConnectorBuilder;
+use log::debug;
+use std::{env, io::Write};
+
+use datadog_trace_protobuf::pb;
+
+const STATS_INTAKE_URL: &str = "https://trace.agent.datadoghq.com/api/v0.2/stats";
+
+pub async fn get_stats_from_request_body(body: Body) -> anyhow::Result<pb::ClientStatsPayload> {
+    let buffer = hyper::body::aggregate(body).await?;
```

Comment:
We're not checking or constraining the size of the body here. 
A broken, oversized, or malicious request will cause an out of memory error.

Reply:
Added a check to make sure the `content-length` header is less than a config set value (10MB). Hyper will stop reading request bodies once `content-length` is reached